test/process_instances/retry_process_instance.spec.ts

```diff
@@ -0,0 +1,152 @@
+      processDefinitionIdRetryProcessInstanceFinish]);
+  });
+
+  after(async (): Promise<void> => {
+    await testFixtureProvider.tearDown();
+  });
+
+  it('should retry a ProcessInstance and finish it', async (): Promise<void> => {
+    const result = await processInstanceHandler.startProcessInstanceAndReturnResult(processDefinitionIdRetryProcessInstance);
+    const {processInstanceId} = result;
+
+    await new Promise(async (resolve) => {
+      processInstanceHandler.waitForProcessWithInstanceIdToWalkInError(processInstanceId, async () => {
+        const processInstance = await (await testFixtureProvider
+          .adminDomainClient
+          .processInstances
+          .query({processInstanceId: processInstanceId})).processInstances[0];
+
+        should(processInstance.state).be.equal('error');
+
+        resolve();
+      });
+    });
+
+    await processInstanceHandler.wait(2000);
+
+    try {
+      await testFixtureProvider.adminDomainClient.processInstances.retryProcessInstance(processInstanceId, superAdmin);
+    } catch (error) {
+      console.error(error);
+    }
+
+    await new Promise(async (resolve) => {
+      processInstanceHandler.waitForProcessWithInstanceIdToEnd(processInstanceId, async () => {
+        const retriedProcessInstance = await (await testFixtureProvider
+          .adminDomainClient
+          .processInstances
+          .query({processInstanceId: processInstanceId})).processInstances[0];
+
+        should(retriedProcessInstance.state).be.equal('finished');
+
+        resolve();
+      });
+    });
+  });
+
+  it('should fail to retry a ProcessInstance with wrong identity', async (): Promise<void> => {
+
+    const result = await processInstanceHandler.startProcessInstanceAndReturnResult(processDefinitionIdRetryProcessInstance);
+    const {processInstanceId} = result;
+
+    try {
+      await testFixtureProvider.adminDomainClient.processInstances.retryProcessInstance(processInstanceId, restrictedIdentity);
+
+      should.fail('result', undefined, 'This should have failed, because the user does not have the required claim');
+
+    } catch (error) {
+      should(error.message).match(/access denied/i);
+      should(error.code).be.equal(403);
+    }
+
+  });
+
+  it('should fail to retry a ProcessInstance which is already finished', async (): Promise<void> => {
+    const result = await processInstanceHandler.startProcessInstanceAndReturnResult(processDefinitionIdRetryProcessInstanceFinish);
+    const {processInstanceId} = result;
+
+    await new Promise(async (resolve) => {
+
+      processInstanceHandler.waitForProcessWithInstanceIdToEnd(processInstanceId, async () => {
+
+        const processInstance = await (await testFixtureProvider
+          .adminDomainClient
+          .processInstances
+          .query({processInstanceId: processInstanceId})).processInstances[0];
+
+        should(processInstance.state).be.equal('finished');
+        resolve();
+      });
+
+    });
```

Comment:
Warum so kompliziert?

```suggestion
    const result = await testFixtureProvider
      .coreDomainAdapter
      .processModels
      .startProcessInstanceAndAwaitEndEvent({
        processModelId: processDefinitionIdRetryProcessInstanceFinish,
      }, identityToUse);
    
    const {processInstanceId} = result;
```

Reply:
Sowas hab ich gesucht, danke 👍